traitlets/config/application.py

```diff
@@ -763,7 +768,8 @@ def load_config_file(self, filename, path=None):
             raise_config_file_errors=self.raise_config_file_errors,
         ):
             new_config.merge(config)
-            self._loaded_config_files.append(filename)
+            if filename not in self._loaded_config_files:  # only add if not present (support reloads)
```

Comment:
The comment throws me off a bit, as reloading a config here just makes it not re-append it to the loaded config files. Would it make more sense to have it read "# only include in list of loaded files if not already loaded"